solutions/secure/variables.tf

```diff
@@ -0,0 +1,177 @@
+        name  = string
+        value = string
+    }))) }))
+    enforcement_mode = string
+    tags = optional(list(object({
+      name  = string
+      value = string
+    })), [])
+    operations = optional(list(object({
+      api_types = list(object({
+        api_type_id = string
+      }))
+    })))
+  }))
+  description = "(Optional, list) List of CBR rule to create for the instance"
+  default     = []
+  # Validation happens in the rule module
+}
+
+variable "bucket_configs" {
+  type = list(object({
+    access_tags                   = optional(list(string), [])
+    bucket_name                   = string
+    kms_encryption_enabled        = optional(bool, true)
+    kms_guid                      = optional(string, null)
+    kms_key_crn                   = string
+    skip_iam_authorization_policy = optional(bool, false)
+    management_endpoint_type      = string
+    cross_region_location         = optional(string, null)
+    storage_class                 = optional(string, "smart")
+    region_location               = optional(string, null)
+    resource_instance_id          = optional(string, null)
+    force_delete                  = optional(bool, true)
+    single_site_location          = optional(string, null)
+    hard_quota                    = optional(number, null)
+
+    activity_tracking = optional(object({
+      read_data_events     = optional(bool, true)
+      write_data_events    = optional(bool, true)
+      activity_tracker_crn = optional(string, null)
+    }))
+    archive_rule = optional(object({
+      enable = optional(bool, false)
+      days   = optional(number, 20)
+      type   = optional(string, "Glacier")
+    }))
+    expire_rule = optional(object({
+      enable = optional(bool, false)
+      days   = optional(number, 365)
+    }))
+    metrics_monitoring = optional(object({
+      usage_metrics_enabled   = optional(bool, true)
+      request_metrics_enabled = optional(bool, true)
+      metrics_monitoring_crn  = optional(string, null)
+    }))
+    object_versioning = optional(object({
+      enable = optional(bool, false)
+    }))
+    retention_rule = optional(object({
+      default   = optional(number, 90)
+      maximum   = optional(number, 350)
+      minimum   = optional(number, 90)
+      permanent = optional(bool, false)
+    }))
+    cbr_rules = optional(list(object({
+      description = string
+      account_id  = string
+      rule_contexts = list(object({
+        attributes = optional(list(object({
+          name  = string
+          value = string
+      }))) }))
+      enforcement_mode = string
+      tags = optional(list(object({
+        name  = string
+        value = string
+      })), [])
+      operations = optional(list(object({
+        api_types = list(object({
+          api_type_id = string
+        }))
+      })))
+    })), [])
+
+  }))
+  description = "Cloud Object Storage bucket configurations"
+  default     = []
+  # Validation happens in the fscloud module
```

Comment:
same comment as above

Reply:
To provide information regarding the validation being taken care of in the FSCloud module of COS.